api_v3/services/SchemaService.php

```diff
@@ -39,20 +39,27 @@ function serveAction($type)
 		header("Content-Type: text/plain; charset=UTF-8");
 		
 		$cacheXsdFile = self::getSchemaPath($type);
-		return $this->dumpFile($cacheXsdFile);
+		return $this->dumpFile($cacheXsdFile, 'application/xml');
 	}
 	
 	/**
-	 * @param KalturaSchemaType $type  
-	 * @return string filePath 
+	 * @param KalturaSchemaType $type
+	 * @return string filePath
 	 */
 	public static function getSchemaPath($type)
 	{
 		$cacheXsdFile = kConf::get("cache_root_path") . "/$type.xsd";
 		if(file_exists($cacheXsdFile))
 			return realpath($cacheXsdFile);
 		
-		$xsdFile = fopen($cacheXsdFile, 'w');
+		$resultXsd = self::buildSchemaByType($type);
+		
+		kFile::safeFilePutContents($cacheXsdFile, $resultXsd, 0644);
```

Comment:
Show validate return value from safeFilePutContents and throw exception on failure.